Design/Packaging/Future_packaging_strategy.md

```diff
@@ -0,0 +1,70 @@
+```bash
+conda create -p package_dir --use-local mantid-workbench --copy -c conda-forge -y
+```
+
+For Linux operating systems I suggest using FPM (Effin Package Management - https://github.com/jordansissel/fpm) due to it's simplicity and ease of creating .deb and .rpm installers, from one machine, therefore in the future it should be possible to create all Linux distribution packages from one machine. At present it is not planned to use FPM for MacOSX as it doesn't support .dmg app bundles, and the support it does provide for .pkg files on MacOSX is less than ideal for the same reasons we already migrated away from .pkg files in the first place.
+
+For MacOSX, we need to make .dmg files, we can use a very similar strategy to linux here but with hdiutil (Disk Utility built into MacOS used similarly to how FreeCAD has: https://github.com/FreeCAD/FreeCAD-Bundle/blob/master/conda/osx/create_bundle.sh)
+
+For Windows, we intend to continue using NSIS to produce a .exe installer. We evaluated WIX unfortunately the amount of time it would take to not only switch but to learn how to switch from NSIS is not worth the benefits, this isn't out of the question long term. 
+
+
+## An overall look at the process
+Everything will be achieved from a single script that can be worked into CMake or some other method of calling. The packaging script should return either a locally built Conda-build that is then uploaded or not uploaded, depending on this upload status we then create the environment by either downloading the dependencies and mantid workbench, or just the dependencies using a local mantid workbench constructed with Conda-build. I have produced a Sequence Diagram to better explain the suggested process:
+![packaging_plan_diagram](packaging_plan_diagram.png)
+
+## Alternatives
+
+The various Linux distributions have their built in .deb and .rpm creation tools, as well as the abstraction layer that comes from CPack enabling building and linking to system packages. We do not want to link to the system packages, whilst conda-build will get all the required packages for us whilst CPack is not capable of this.
+
+MacOSX currently uses CPack, it has a custom ruby script and would likely work for our current purposes, however it is frustrating to maintain, we would be able to unite the script for package creation for MacOSX and POSIX systems and a lot of the details should allow for easier packaging scripts. 
+
+Windows at present uses CPack and NSIS to create the packaging. With CPack being rolled back, we intend to stick with NSIS via the larger packaging script. At present we just put in the required dependency .dll by installing them into the bin of the package, Conda will handle this for us.
+
+## Looking to the future
+Considering Mantid workbench is now approaching behemouth status like Mantid Plot, it may be prudent to look at potential future project deliverables that don't include all parts of workbench that are included now.
```

Comment:
More of a nit, but I'm not sure on calling Workbench a behemoth like Plot, as Plot was unmaintainable from no tests and god classes. Workbench, which is large but well tested, isn't a "behemoth" and might give the wrong ideas to readers. 

Could we use something like, "Mantid Workbench is large and all-encompassing" or similar?

Reply:
I agree. The core functionality of Workbench is far reduced from MantidPlot and Workbench is far more modular and maintainable. I see the all-encompassing thing as the `-all` package being a bit like `Anaconda` itself - it contains everything that you might need and it easy for users to know what they are getting. There are also good reasons for providing the other, more minimal, package types but it's not because Workbench  itself is too big.

Reply:
Yes that makes more sense to me I will change it.